x/posts/internal/types/post.go

```diff
@@ -36,28 +36,53 @@ func ParsePostID(value string) (PostID, error) {
 	return PostID(intVal), err
 }
 
+// ----------------
+// --- Post IDs
+// ----------------
+
+// PostIDs represents a slice of PostID objects
+type PostIDs []PostID
+
+// Equals returns true iff the ids slice and the other
+// one contain the same data in the same order
+func (ids PostIDs) Equals(other PostIDs) bool {
+	if len(ids) != len(other) {
+		return false
+	}
+
+	for index, id := range ids {
+		if id != other[index] {
+			return false
+		}
+	}
+
+	return true
+}
+
 // ---------------
 // --- Post
 // ---------------
 
 // Post is a struct of a Magpie post
 type Post struct {
-	PostID     PostID         `json:"id"`
-	ParentID   PostID         `json:"parent_id"`
-	Message    string         `json:"message"`
-	Created    int64          `json:"created"`     // Block height at which the post has been created
-	LastEdited int64          `json:"last_edited"` // Block height at which the post has been edited the last time
-	Owner      sdk.AccAddress `json:"owner"`
+	PostID         PostID         `json:"id,string"`
+	ParentID       PostID         `json:"parent_id,string"`
+	Message        string         `json:"message"`
+	Created        sdk.Int        `json:"created"`     // Block height at which the post has been created
+	LastEdited     sdk.Int        `json:"last_edited"` // Block height at which the post has been edited the last time
+	AllowsComments bool           `json:"allows_comments"`
+	Owner          sdk.AccAddress `json:"owner"`
 }
 
-func NewPost(ID, parentID PostID, message string, created int64, owner sdk.AccAddress) Post {
+func NewPost(id, parentID PostID, message string, allowsComments bool, created int64, owner sdk.AccAddress) Post {
```

Comment:
I think we better follow the rule. https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
Actually, by running the `make lint` command, if we change `id` to `ID`, the following error will be shown up: 

```
x/posts/internal/types/post.go:77:14: captLocal: `ID' should not be capitalized (gocritic)
func NewPost(ID, parentID PostID, message string, allowsComments bool, created int64, owner sdk.AccAddress) Post {
             ^
```

I think we should following coding rules that can be automatically checked from [GolangCI](https://golangci.com/) (which we should enable for this repo as the config file already exists)